universe/Pathfinder.cpp

```diff
@@ -1094,7 +1094,14 @@ std::pair<std::vector<int>, int> Pathfinder::LeastJumpsPath(
 std::pair<std::vector<int>, int> Pathfinder::PathfinderImpl::LeastJumpsPath(
     int system1_id, int system2_id, int empire_id, int max_jumps) const
 {
-    if (empire_id == ALL_EMPIRES) {
+    bool use_system_graph = false;
+    auto empire_view_graph_it = m_graph_impl.empire_system_graph_views.find(empire_id);
+    if (empire_id == ALL_EMPIRES || empire_view_graph_it == m_graph_impl.empire_system_graph_views.end()) {
```

Comment:
I'm slightly concerned that there could be situations where this somehow lets a player use unknown lanes, but it's probablz better than not at all.workimg pathing...

Reply:
If there is a different way to figure out we are on client/AI it would probably be more robust to use that instead of falling back and hoping for the best.

Reply:
ok, so lets keep the fallback.
For the client fallback let us use the system graph. The ALL_EMPIRES view could be used for neutral/monsters

What i would want:
* on server if there is no empire specific graph -> error
* on client if there is no empire specific graph -> use system_graph

as i dont see how to distinguish betwenn client/server in a good way; i would implement like this:
* on server if there is no empire specific graph -> use system_graph
* on client if there is no empire specific graph -> use system_graph

Reply:
It looks strange if the server doesn't have a graph of the existing empire and the client doesn't have a graph of own empire.